packages/bpk-component-infinite-scroll/src/withInfiniteScroll.js

```diff
@@ -113,108 +114,111 @@ const withInfiniteScroll = (
         showSeeMore: false,
       };
 
-      this.onItemsFetch = props.onItemsFetch;
-      this.observer = new IntersectionObserver(this.handleIntersection, {
-        threshold: 1,
-      });
-    }
-
-    componentDidMount() {
-      if (this.sentinel) {
-        this.observer.observe(this.sentinel);
-      }
+      this.props.dataSource.onDataChange(this.updateData);
     }
 
-    componentDidUpdate() {
-      if (this.sentinel && this.state.index > 0) {
-        this.observer.observe(this.sentinel);
-      }
+    componentWillMount() {
+      this.loadMore().then(newState => {
+        this.setState(newState);
+      });
     }
 
-    componentWillUnMount() {
-      if (this.sentinel) {
-        this.observer.unobserve(this.sentinel);
-      }
-    }
+    updateData = () => {
```

Comment:
That was not how I was thinking about this tbh. I think `updateData` should receive all the new data from the `DataSource` and the callback here would only set the state

Reply:
but in the dataset we don't have knowledge about current index and number of elements per scroll.

Reply:
True. I'm thinking if it wouldn't be better to have this info in the datasource then. I.e. the component will only call `dataSource.fetchMore()` and then `elementsPerScroll` and `index` will be part of the dataSource state.

For example: `new ArrayDataSource(data, { pageSize: 10 })`.

What do you think?

Reply:
what about 
```
      this.props.dataSource.onDataChange((updatedElements: Array<any>) => {
        const { index } = this.state;
        const { elementsPerScroll } = this.props;
        this.setState({
          elementsToRender: updatedElements.slice(
            index - elementsPerScroll,
            elementsPerScroll),
        });
      });
```
I prefer to keep the in the Infinite Scroll component instead of moving it to DataSource (to maintain it small/simple). However moving this to DataSource may allow us to reset the index as we want.